src/main/java/org/skife/jdbi/v2/StatementContext.java

```diff
@@ -55,6 +57,14 @@
      */
     Map<String, Object> getAttributes();
 
+    /**
+     * Obtain a column mapper for the given type in this context.
+     *
+     * @param type the target type to map to
+     * @return a ResultColumnMapper for the given type
+     * @throws org.skife.jdbi.v2.exceptions.DBIException if no mappers are registered for the given type.
+     */
+    ResultColumnMapper columnMapperFor(Class type);
```

Comment:
Yes. In hindsight, the ResultSetMapperFactory should have probably also taken a Type instead of Class, but I guess @brianm was young and needed the money. :-)

This is a case of "will need retrofitting in a later version". But as we are stuck with our public API, we will have to wait for JDBIv3.